### PR TITLE
redmine 5702 - switch to high level IPv4 functions instead of low level ip2long32() etc 

### PR DESCRIPTION
**Pull request by stilez**

### CODE REVIEW

File: src/usr/local/www/services_dhcp.php [hunk 1]
```diff
@@ -400,14 +400,14 @@ function validate_partial_mac_list($maclist) {
...
+			$subnet_start = gen_subnetv4($ifcfgip, $ifcfgsn);  
+ 			$subnet_end = gen_subnetv4_max($ifcfgip, $ifcfgsn);  
 
-			if (ip2ulong($_POST['range_from']) > ip2ulong($_POST['range_to'])) {
+			if (ip_greater_than($_POST['range_from'], $_POST['range_to'])) {  
 				$input_errors[] = gettext("The range is invalid (first element higher than second element).");
 			}
 
-			if (ip2ulong($_POST['range_from']) < $subnet_start || ip2ulong($_POST['range_to']) > $subnet_end) {
+			if (!is_inrange_v4($_POST['range_from'], $subnet_start, $subnet_end)) {
```
Comment (rbgarga) [inline src/usr/local/www/services_dhcp.php]: Please remove spaces from EOL

File: src/usr/local/www/services_dhcp.php [hunk 2]
```diff
@@ -400,14 +400,14 @@ function validate_partial_mac_list($maclist) {
 
 		if (!$input_errors) {
 			/* make sure the range lies within the current subnet */
-			$subnet_start = ip2ulong(long2ip32(ip2long($ifcfgip) & gen_subnet_mask_long($ifcfgsn)));
-			$subnet_end = ip2ulong(long2ip32(ip2long($ifcfgip) | (~gen_subnet_mask_long($ifcfgsn))));
+			$subnet_start = gen_subnetv4($ifcfgip, $ifcfgsn);  
+ 			$subnet_end = gen_subnetv4_max($ifcfgip, $ifcfgsn);  
 
-			if (ip2ulong($_POST['range_from']) > ip2ulong($_POST['range_to'])) {
+			if (ip_greater_than($_POST['range_from'], $_POST['range_to'])) {
```
Comment (rbgarga) [inline src/usr/local/www/services_dhcp.php]: Please remove spaces from EOL

File: src/usr/local/www/services_dhcp.php [hunk 3]
```diff
@@ -400,14 +400,14 @@ function validate_partial_mac_list($maclist) {
 
 		if (!$input_errors) {
 			/* make sure the range lies within the current subnet */
-			$subnet_start = ip2ulong(long2ip32(ip2long($ifcfgip) & gen_subnet_mask_long($ifcfgsn)));
-			$subnet_end = ip2ulong(long2ip32(ip2long($ifcfgip) | (~gen_subnet_mask_long($ifcfgsn))));
+			$subnet_start = gen_subnetv4($ifcfgip, $ifcfgsn);  
+ 			$subnet_end = gen_subnetv4_max($ifcfgip, $ifcfgsn);
```
Comment (rbgarga) [inline src/usr/local/www/services_dhcp.php]: Please remove spaces from EOL

Comment (rbgarga) [inline src/usr/local/www/services_dhcp.php]: There is also an extra space at the begin of line

**Comment by stilez** [comment]
Note - I think there are other logic errors in this one. Will fix once PR #2421 is dealt with (which I'll use to fix it)

**Comment by rbgarga** [comment]
Extra spaces at EOL are still present

**Comment by stilez** [comment]
Can you hold off committing this one until #2421 is committed (if it's ok now) and I can use it to fix what looks like a related logic issue?

**Comment by rbgarga** [comment]
@stilez let me know when it's OK, #2421 was just merged. Thanks!

**Comment by rbgarga** [comment]
Any news about this one @stilez ?

File: src/usr/local/www/services_dhcp.php [hunk 4]
```diff
@@ -400,14 +400,14 @@ function validate_partial_mac_list($maclist) {
...
+			$subnet_start = gen_subnetv4($ifcfgip, $ifcfgsn);
+			$subnet_end = gen_subnetv4_max($ifcfgip, $ifcfgsn);
 
-			if (ip2ulong($_POST['range_from']) > ip2ulong($_POST['range_to'])) {
+			if (ip_greater_than($_POST['range_from'], $_POST['range_to'])) {
 				$input_errors[] = gettext("The range is invalid (first element higher than second element).");
 			}
 
-			if (ip2ulong($_POST['range_from']) < $subnet_start || ip2ulong($_POST['range_to']) > $subnet_end) {
+			if (!is_inrange_v4($_POST['range_from'], $subnet_start, $subnet_end)) {
```
Comment (rbgarga) [inline src/usr/local/www/services_dhcp.php]: You should check $_POST['range_to'] to make sure it's inside the current subnet, no?

File: src/usr/local/www/services_dhcp.php [hunk 5]
```diff
@@ -435,15 +435,12 @@ function validate_partial_mac_list($maclist) {
...
-			$dynsubnet_start = ip2ulong($_POST['range_from']);
-			$dynsubnet_end = ip2ulong($_POST['range_to']);
 			if (is_array($a_maps)) {
 				foreach ($a_maps as $map) {
 					if (empty($map['ipaddr'])) {
 						continue;
 					}
-					if ((ip2ulong($map['ipaddr']) >= $dynsubnet_start) &&
-					    (ip2ulong($map['ipaddr']) <= $dynsubnet_end)) {
+					if (is_inrange_v4($map['ipaddr'], $_POST['range_from'], $_POST['range_to'])) {
```
Comment (rbgarga) [inline src/usr/local/www/services_dhcp.php]: Are you going to move this check to overlap related functions merged in #2421?

**Pull request merged**